commonmark/src/main/java/org/commonmark/internal/DocumentParser.java

```diff
@@ -59,11 +73,17 @@ public DocumentParser(List<BlockParserFactory> blockParserFactories, InlineParse
         activateBlockParser(this.documentBlockParser);
     }
 
-    public static List<BlockParserFactory> calculateBlockParserFactories(List<BlockParserFactory> customBlockParserFactories) {
+    public static List<Class<? extends Block>> getDefaultBlockParserTypes() {
+        return CORE_FACTORY_TYPES;
+    }
+
+    public static List<BlockParserFactory> calculateBlockParserFactories(List<BlockParserFactory> customBlockParserFactories, List<Class<? extends Block>> allowedBlockTypes) {
         List<BlockParserFactory> list = new ArrayList<>();
         // By having the custom factories come first, extensions are able to change behavior of core syntax.
         list.addAll(customBlockParserFactories);
```

Comment:
I was a bit hesitant of this change because it didn't handle custom block parsers coming from extensions. But then again, the library user is the one actually configuring the extensions, so I don't think this is a problem :).